common/logger/src/struct_log.rs

```diff
@@ -476,8 +475,20 @@ impl TCPStructLogThread {
         }
     }
 
-    pub fn connect(&self) -> io::Result<TcpStream> {
-        TcpStream::connect_timeout(&self.address, Duration::from_millis(CONNECTION_TIMEOUT_MS))
+    pub fn connect(&mut self) -> io::Result<TcpStream> {
+        // resolve addresses to handle DNS names
+        for addr in self.address.to_socket_addrs()? {
+            if let Ok(stream) =
+                TcpStream::connect_timeout(&addr, Duration::from_millis(CONNECTION_TIMEOUT_MS))
+            {
+                return Ok(stream);
+            }
+        }
+
+        Err(io::Error::new(
+            io::ErrorKind::Other,
+            format!("Unable to resolve and connect to {}", self.address),
```

Comment:
Will this show the actual connection error?

Reply:
ahhh good, call, we should probably take in the result and check if it is okay, if it is not, we can return the last error

Reply:
No, it doesn't show the actual connection error.

As something David mentioned before that one could be IPv6 and one could be Ipv4, and we could run through a bunch, I probably should merge the errors together

Reply:
Just getting the last error is fine, in k8s the address will be v4 only.

Reply:
Fixed to retrun the last error.